### PR TITLE
Add Zero-copy migrations feature page

### PR DESCRIPTION
**Pull request by mz-pdm**
Feature

Related to issue https://bugzilla.redhat.com/show_bug.cgi?id=2089434

Changes proposed in this pull request:

- Feature page for zero-copy migrations added.

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/main/CONTRIBUTING.md): @mz-pdm 

This pull request needs review by: @michalskrivanek @mwperina

### CODE REVIEW

File: source/develop/release-management/features/virt/zerocopy-migrations.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Nothing special, just up-to-date libvirt and QEMU.
+
+## Limitations
+
+Current implementation of zero-copy migration in QEMU has some restrictions:
+
+- It can be used only with [parallel migrations](parallel-migrations.html) and thus the same limitations as for parallel migrations apply.
+
+- It cannot be used with migration encryption.
```
Comment (michalskrivanek) [inline source/develop/release-management/features/virt/zerocopy-migrations.md:34]: and parallel migrations alone do work with encryption?

File: source/develop/release-management/features/virt/zerocopy-migrations.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+- QEMU must be permitted to lock guest RAM pages in memory (libvirt takes care of this unless memory limits are set manually in ``<memtune>``.
+
+- This is a new feature in QEMU/libvirt and may contain bugs.  It's not recommended to use it in oVirt at the moment it unless it is really needed.
+
+- Vdsm, libvirt and QEMU on the migration source host must support the feature. 
+
+## User experience
+
+A new migration policy *Very large VMs* is introduced in cluster migration options.  If this policy is selected, zero-copy migrations are conditionally enabled and other migrations parameters specified in the policy are applied (this allows using migrations parameters such as downtime schedule customized for such migrations).  The policy is available only when parallel migrations are enabled in Engine config (i.e. on cluster level 4.7 by default).
```
Comment (michalskrivanek) [inline source/develop/release-management/features/virt/zerocopy-migrations.md:44]: "conditionally"?

File: source/develop/release-management/features/virt/zerocopy-migrations.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+- If parallel migrations are disabled or only conditionally enabled ("Auto") then they are enabled with automatic number of parallel connections.
+
+- If migration encryption is enabled then zero-copy is not enabled.
+
+If the policy is selected and the host doesn't support the feature then:
+
+- If Vdsm doesn't support the feature, API call mismatch is logged and zero-copy is not enabled.
+
+- If Vdsm supports the feature but libvirt doesn't then zero-copy is not enabled.
```
Comment (michalskrivanek) [inline source/develop/release-management/features/virt/zerocopy-migrations.md:56]: I don't think you have to handle this case. You can bump requirements as part of this feature

File: source/develop/release-management/features/virt/zerocopy-migrations.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+This applies only when parallel migrations are enabled in Engine config (cluster level 4.7 by default) and the host is not up-to-date.
+
+In all the cases, VM migration proceeds, although with different parameters.
+
+## Testing
+
+It should be tested that when the policy above is selected and the limitations above are satisfied then ``VIR_MIGRATE_ZEROCOPY`` flag is passed to the libvirt migration call.
```
Comment (michalskrivanek) [inline source/develop/release-management/features/virt/zerocopy-migrations.md:64]: so add that to OST and you're done....

File: source/develop/release-management/features/virt/zerocopy-migrations.md (hunk 1 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/zerocopy-migrations.md:34]: Yes.

File: source/develop/release-management/features/virt/zerocopy-migrations.md (hunk 4 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/zerocopy-migrations.md:64]: Added to my to-do list.

File: source/develop/release-management/features/virt/zerocopy-migrations.md (hunk 2 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/zerocopy-migrations.md]: Clarified.

File: source/develop/release-management/features/virt/zerocopy-migrations.md (hunk 3 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/zerocopy-migrations.md]: Right, the feature should reach RHEL and CentOS Stream 8 eventually.  Removed.

**Comment by mz-pdm** [comment]
> 1 failing checks

Looks like an unrelated permission problem.

**Comment by mz-pdm** [comment]
Added a remark about downtimes and autoconverge.

**Comment by mz-pdm** [comment]
> Looks like an unrelated permission problem.

The same failure again.

**Review by michalskrivanek — APPROVED** [review]

**Pull request merged**